osu.iOS/AppDelegate.cs

```diff
@@ -0,0 +1,31 @@
+﻿// Copyright (c) 2007-2018 ppy Pty Ltd <contact@ppy.sh>.
+// Licensed under the MIT Licence - https://raw.githubusercontent.com/ppy/osu/master/LICENCE
+using Foundation;
+using osu.Framework.iOS;
+using osu.Game;
+using osu.Game.Beatmaps;
+using osu.Framework.Allocation;
+
+namespace osu.iOS
+{
+    // The UIApplicationDelegate for the application. This class is responsible for launching the
+    // User Interface of the application, as well as listening (and optionally responding) to application events from iOS.
+
+    [Register("AppDelegate")]
+    public class AppDelegate : GameAppDelegate
+    {
+        OsuGame iOSOsuGame = new OsuGame();
+
+        protected override Framework.Game CreateGame()
+        {
+           return iOSOsuGame;
+        }
+
+        [Export("application:openURL:options:")]
+        override public bool OpenUrl(UIKit.UIApplication app, NSUrl url, NSDictionary options)
```

Comment:
Well I did write the initial abstract method in the framework [ppy/osu-framework#2048](https://github.com/ppy/osu-framework/pull/2048) and I explained ppy the need for it.

I don't see how I would be able to call OsuGame.Import all the way back in the framework when the framework does not contain that class itself.  iOS handles its OpenUrl methods through the AppDelegate when you want to open Documents/Files etc from other apps. That is the main entry point. 

I mainly just ported this over from its [desktop counterpart](https://github.com/ppy/osu/blob/master/osu.Desktop/OsuGameDesktop.cs#L77) that has this implemented in the game itself too

Reply:
The desktop counterpart extends `OsuGame` to `OsuGameDesktop` to handle the `FileDrop` event.  It would be nicer to make `OsuGameIOS` do similar, and invoke the `FileDrop` event from `OpenUrl` in osuTK.

Reply:
I see! I will look into that tomorrow. Thanks for your feedback!